### PR TITLE
Compile generic functions at call-time

### PR DESCRIPTION
**Pull request by ManOfTeflon**
Summary: For functions with type parameters, compilation must be
deferred until the function in instantiated, rather than being performed
when the schema is loaded as with ordinary functions.  To accomplish
this, a new expression type is introduced, which tracks the uncompiled
AST for use at call time.

### CODE REVIEW

**Comment by ManOfTeflon** [comment]
Current dependencies on/for this PR:
* main
  * **PR #49** <a href="https://app.graphite.dev/github/pr/qscl/queryscript/49" target="_blank">Graphite</a> 
    * **PR #50** <a href="https://app.graphite.dev/github/pr/qscl/queryscript/50" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/qscl/queryscript/50?utm_source=stack-comment).

File: queryscript/src/compile/schema.rs [hunk 1]
```diff
@@ -819,6 +819,7 @@ where
     ContextRef(Ident),
     Connection(Arc<ConnectionString>),
     Materialize(MaterializeExpr<TypeRef>),
+    UncompiledFn(ast::FnDef),
```
Comment (ankrgyl) [inline queryscript/src/compile/schema.rs:822]: not something we should do in this change but another good piece of evidence that we should have a separate runtime expr tree i think

File: queryscript/src/compile/sql.rs [hunk 2]
```diff
@@ -2408,8 +2411,22 @@ pub fn compile_sqlexpr(
                     }
 
                     let func_expr = func.expr.unwrap_schema_entry().await?;
+                    let compiled_func_expr = match func_expr {
+                        Expr::UncompiledFn(def) => {
+                            let (compiled_body, _) = compile_fn_body(
```
Comment (ankrgyl) [inline queryscript/src/compile/sql.rs:2416]: should we assert something about the generics here? I assume they should be empty?

**Review by ankrgyl — APPROVED** [review]

**Comment by ManOfTeflon** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: 7vyZp84pP13YDKIYE5Y0.
See details on [graphite.dev](https://app.graphite.dev/github/pr/qscl/queryscript/50)

**Pull request merged**